### PR TITLE
[identity-880] Move retry decorator out and add custom retry counts

### PR DESCRIPTION
**Pull request by coffeyk**
Add the ability to customize the retry counts on a per function basis.

Metrics on fails and retries
black formatting

This has been tested inside a docker dev environment locally. Will test in staging once I mint a new release.

### CODE REVIEW

File: beaker_extensions/cassandra_cql.py [hunk 1]
```diff
@@ -1,28 +1,93 @@
...
+    _tries = 0  # type: int
+    _RETRYABLE_EXCEPTIONS = tuple()  # type: tuple[Type[Exception]]
+
+
+def _retry(
+    func=None,  # type: Optional[Callable]
+    tries=None,  # type: Optional[int]
+):
+    if func is None:
+        return partial(_retry, tries=tries)
```
Comment (coffeyk) [inline beaker_extensions/cassandra_cql.py:47]: This allows _retry to either be called as
```
@_retry
def func(): ...

# or
@_retry(tries=3)
def func(): ...
```

File: beaker_extensions/cassandra_cql.py [hunk 2]
```diff
@@ -81,9 +160,14 @@ class _CassandraBackedDict(object):
...
 
-    _RETRYABLE_EXCEPTIONS = (cassandra.DriverException, cassandra.RequestExecutionException)
+    _RETRYABLE_EXCEPTIONS = (
+        cassandra.DriverException,
+        cassandra.RequestExecutionException,
+    )
 
-    def __init__(self, namespace, url=None, keyspace=None, column_family=None, **params):
+    def __init__(
+        self, namespace, url=None, keyspace=None, column_family=None, **params
```
Comment (sseg-dd) [inline beaker_extensions/cassandra_cql.py:169]: It wasn't well documented before, but is there any usage of `params.tries` in our use of the library (setting `self._tries` about 10 lines below here)?

Comment (sseg-dd) [reply beaker_extensions/cassandra_cql.py:169]: Or maybe this class isn't used outside of this package, in which case we could maybe remove the data member `self._tries`?

Comment (sseg-dd) [reply beaker_extensions/cassandra_cql.py:169]: oh it's still in the wrapper!

**Review by sseg-dd — APPROVED** [review]

File: beaker_extensions/cassandra_cql.py (hunk 2 above)
Comment (coffeyk) [reply beaker_extensions/cassandra_cql.py:194]: Yeah, the `_retry` decorator expects the wrapped functions to belong to a class with `_tries` and `_RETRYABLE_EXCEPTIONS` attributes.

**Pull request merged**